### DISCUSSION TITLE
What's the difference between result.map and result.try?

### DISCUSSION THREAD

**Discussion by suzaku** [gleam-lang/gleam#2718] [General]
I'm reading the language tour and the description of these two read almost identical, what's the difference?

**Comment by schurhammer** [comment]
`map` is used to apply a function if the result is `Ok`
`try` is similar but it is used when that function also returns a result. It is essentially `map` + `flatten`.
```gleam
// using map to add one
result.map(Ok(1), fn(x) { x + 1 }) // Ok(2)

// if the map returns a result, we get nested results
result.map(Ok(1), fn(x) { Ok(x + 1) }) // Ok(Ok(2))

// we can use try to avoid this nesting
result.try(Ok(1), fn(x) { Ok(x + 1) }) // Ok(2)
```
